pkg/controller/templateupdaterequest/templateupdaterequest_controller_test.go

```diff
@@ -0,0 +1,445 @@
+
+				// check that TemplateUpdateRequest is in "updating" condition
+				turtest.AssertThatTemplateUpdateRequest(t, "user-1", cl).
+					HasConditions(toBeUpdating()).
+					HasSyncIndexes(map[string]string{
+						"cluster1": "1",
+					})
+			})
+
+			t.Run("with custom template when cluster resources exist in update", func(t *testing.T) {
+				// given
+				initObjs := []runtime.Object{&newNSTemplateTier}
+				mur := murtest.NewMasterUserRecord(t, "user-1",
+					murtest.Account("cluster1", oldNSTemplateTier, murtest.SyncIndex("1"),
+						murtest.CustomClusterResourcesTemplate("custom"), // a custom template is defined for the 1st namespace
+					))
+				initObjs = append(initObjs, mur)
+				initObjs = append(initObjs, turtest.NewTemplateUpdateRequest("user-1", newNSTemplateTier))
+				r, req, cl := prepareReconcile(t, "user-1", initObjs...)
+				// when
+				res, err := r.Reconcile(req)
+				// then
+				require.NoError(t, err)
+				require.Equal(t, reconcile.Result{}, res) // no need to requeue, the MUR is watched
+				// check that the MasterUserRecord was updated
+				murtest.AssertThatMasterUserRecord(t, "user-1", cl).
+					AllUserAccountsHaveTier(newNSTemplateTier).
+					// check that the custom template for the given cluster/namespace was not lost
+					HasCustomClusterResourcesTemplate("cluster1", "custom")
+
+				// check that TemplateUpdateRequest is in "updating" condition
+				turtest.AssertThatTemplateUpdateRequest(t, "user-1", cl).
+					HasConditions(toBeUpdating()).
+					HasSyncIndexes(map[string]string{
+						"cluster1": "1",
+					})
+			})
+
+			t.Run("with custom template when cluster resources do not exist in update", func(t *testing.T) {
+				// given
+				initObjs := []runtime.Object{&newNSTemplateTier3}
+				mur := murtest.NewMasterUserRecord(t, "user-1",
+					murtest.Account("cluster1", oldNSTemplateTier, murtest.SyncIndex("1"),
+						murtest.CustomClusterResourcesTemplate("custom"), // a custom template is defined for the 1st namespace
+					))
+				initObjs = append(initObjs, mur)
+				initObjs = append(initObjs, turtest.NewTemplateUpdateRequest("user-1", newNSTemplateTier3))
+				r, req, cl := prepareReconcile(t, "user-1", initObjs...)
+				// when
+				res, err := r.Reconcile(req)
+				// then
+				require.NoError(t, err)
+				require.Equal(t, reconcile.Result{}, res) // no need to requeue, the MUR is watched
+				// check that the MasterUserRecord was updated
+				murtest.AssertThatMasterUserRecord(t, "user-1", cl).
+					AllUserAccountsHaveTier(newNSTemplateTier3).
+					// check that the custom template for the given cluster/namespace was not lost
+					HasCustomClusterResourcesTemplate("cluster1", "custom")
+
+				// check that TemplateUpdateRequest is in "updating" condition
+				turtest.AssertThatTemplateUpdateRequest(t, "user-1", cl).
+					HasConditions(toBeUpdating()).
+					HasSyncIndexes(map[string]string{
+						"cluster1": "1",
+					})
+			})
+		})
```

Comment:
I think that you haven't covered the case when the ClusterResources template is removed from the tier and no custom template is used:
```suggestion
			t.Run("when cluster resources do not exist in update then it should be removed", func(t *testing.T) {
				// given
				initObjs := []runtime.Object{&newNSTemplateTier3}
				mur := murtest.NewMasterUserRecord(t, "user-1",
					murtest.Account("cluster1", oldNSTemplateTier, murtest.SyncIndex("1")))
				initObjs = append(initObjs, mur)
				initObjs = append(initObjs, turtest.NewTemplateUpdateRequest("user-1", newNSTemplateTier3))
				r, req, cl := prepareReconcile(t, "user-1", initObjs...)
				// when
				res, err := r.Reconcile(req)
				// then
				require.NoError(t, err)
				require.Equal(t, reconcile.Result{}, res) // no need to requeue, the MUR is watched
				// check that the MasterUserRecord was updated
				murtest.AssertThatMasterUserRecord(t, "user-1", cl).
					AllUserAccountsHaveTier(newNSTemplateTier3)

				// check that TemplateUpdateRequest is in "updating" condition
				turtest.AssertThatTemplateUpdateRequest(t, "user-1", cl).
					HasConditions(toBeUpdating()).
					HasSyncIndexes(map[string]string{
						"cluster1": "1",
					})
			})
		})
```

Reply:
wow, that's a nice (and huge) suggestion 😅 thanks